contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,207 @@
+
+> See our blog on [A/B testing mistakes](https://posthog.com/blog/ab-testing-mistakes#7-testing-an-unclear-hypothesis) to read more on how to avoid testing an unclear hypothesis.
+
+### 3. Tests as small of a change as *reasonably* possible 
+
+Ideally, an A/B test should change one thing at a time. This ensures that any change in user behavior can be attributed to the variable that was changed. If multiple variables are changed, it's difficult to determine which one caused the change in behavior.
+
+However, the caveat here is that too small of change can slow your team down. Since running A/B tests take time, if you're constantly testing small changes it may take long to ship large, meaningful changes.
+
+A good rule of thumb to know if your change is too small is if it's unlikely to impact user behavior significantly. To help inform this decision, you can use qualitative data from user research or quantitative data from previous A/B tests.
+
+### 4. Has a sufficiently large sample size of users
+
+A large sample size for your test ensures the *statistical significance* of your experiment results i.e., if you can be confident that your results are not due to chance, but rather a true effect of the changes you made.
+
+To know whether your change is large enough, you'll need to know:
+
+1. Your current conversion rate.
+2. Your "minimum detectable effect" i.e., the smallest change in the conversion rate that you want to detect. The smaller the change, the larger the sample size you'll need.
+3. Your desired level of confidence (the industry standard is 95%).
+
+You then use a [formula](https://en.wikipedia.org/wiki/Sample_size_determination) to determine if your sample size is large enough. There are many calculators online that will do this for you, so you can avoid calculating this yourself (We also include this calculator when creating a new [A/B test in PostHog](/ab-testing/features)).
+
+### 5. Has a long enough test duration
+
+Building on the notion of ensuring statistical significant results, once you've calculated your required sample size, you'll be able to calculate how long your experiment should run for. You do this by dividing your sample size by your daily number of eligible users.
+
+For example, if you're making changes to your signup flow and your required sample size is 1,000 new sign-ups. If your daily number of sign-ups is 100, then you'll need to run your experiment for `1,000 / 100 = 10 days`.
+
+Once again, there are many calculators online to help you determine your duration, and PostHog also includes this calculator when creating a new experiment.
+
+**A good rule of thumb is that your experiment duration should be between one week and one month:
+**
+- One week is a good minimum since users may behave differently on weekends or weekdays.
+- One month is a good maximum since otherwise you may delay shipping important product changes.
+
+## When not to A/B test
```

Comment:
This doesn't feel like a super compelling section right now. Feels a bit like "when an A/B test is the opposite of a good A/B test," so not super necessary.

Reply:
I kind of agree, mainly because this breaks the flow somewhat to the next section, but I think it's still useful info. Maybe we move this into the FAQ section?